### PR TITLE
Update the comments of the method createQueryTo in ActiveRecordFactory so that the IDE can automatically identify the specific class of ActiveRecord

### PR DESCRIPTION
**Pull request by niqingyang**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️/❌
| Breaks BC?    | ✔️/❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Review by xepozz — APPROVED** [review]

**Comment by niqingyang** [comment]
> Also need to check that tests passed

🎉 It has been tests passed

File: src/ActiveQuery.php [hunk 1]
```diff
@@ -107,6 +107,15 @@ class ActiveQuery extends Query implements ActiveQueryInterface
     private array $joinWith = [];
     private ActiveRecordInterface|null $arInstance = null;
 
+    /**
+     * @param class-string $arClass
```
Comment (Tigrov) [inline src/ActiveQuery.php:111]:
```suggestion
     * @param string $arClass
```

Psalm annotations should be specified with `psalm-` prefix, e.g. `@psalm-param`. For `$arClass` it is already exists.
PHPDoc type annotations specified without `psalm-` prefix should be correct, e.g. `@param string`

File: src/ActiveRecordFactory.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ public function __construct(private Factory $factory)
     /**
      * Allows you to create an active record instance through the factory.
      *
-     * @param string $arClass active record class.
+     * @param class-string<T> $arClass active record class.
```
Comment (Tigrov) [inline src/ActiveRecordFactory.php:23]:
```suggestion
     * @param string $arClass active record class.
```

File: src/BaseActiveRecord.php [hunk 3]
```diff
@@ -320,12 +320,15 @@ public function hasAttribute($name): bool
      *
      * Call methods declared in {@see ActiveQuery} to further customize the relation.
      *
-     * @param string $class The class name of the related record
+     * @param class-string<T> $class The class name of the related record
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:323]:
```suggestion
     * @param string $class The class name of the related record
```

File: src/BaseActiveRecord.php [hunk 4]
```diff
@@ -320,12 +320,15 @@ public function hasAttribute($name): bool
...
      * @param array $link The primary-foreign key constraint. The keys of the array refer to the attributes of the
      * record associated with the `$class` model, while the values of the array refer to the corresponding attributes in
      * **this** AR class.
      *
      * @return ActiveQueryInterface The relational query object.
+     *
+     * @psalm-template T of ActiveRecordInterface
+     * @psalm-param class-string<T> $arClass
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:331]:
```suggestion
     * @psalm-param class-string<ActiveRecordInterface> $arClass
```

File: src/BaseActiveRecord.php [hunk 5]
```diff
@@ -356,18 +359,28 @@ public function hasMany(string $class, array $link): ActiveQueryInterface
      *
      * Call methods declared in {@see ActiveQuery} to further customize the relation.
      *
-     * @param string $class The class name of the related record.
+     * @param class-string<T> $class The class name of the related record.
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:362]:
```suggestion
     * @param string $class The class name of the related record.
```

File: src/BaseActiveRecord.php [hunk 6]
```diff
@@ -356,18 +359,28 @@ public function hasMany(string $class, array $link): ActiveQueryInterface
...
      * @param array $link The primary-foreign key constraint. The keys of the array refer to the attributes of the
      * record associated with the `$class` model, while the values of the array refer to the corresponding attributes in
      * **this** AR class.
      *
      * @return ActiveQueryInterface The relational query object.
+     *
+     * @psalm-template T of ActiveRecordInterface
+     * @psalm-param class-string<T> $arClass
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:370]:
```suggestion
     * @psalm-param class-string<ActiveRecordInterface> $arClass
```

File: src/BaseActiveRecord.php [hunk 7]
```diff
@@ -356,18 +359,28 @@ public function hasMany(string $class, array $link): ActiveQueryInterface
...
     {
         return $this->createRelationQuery($class, $link, false);
     }
 
+    /**
+     * @param class-string<T> $arClass
+     * @return ActiveQueryInterface
+     *
+     * @psalm-template T of ActiveRecordInterface
+     * @psalm-param class-string<T> $arClass
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:382]:
```suggestion
     * @psalm-param class-string<ActiveRecordInterface> $arClass
```

File: src/BaseActiveRecord.php [hunk 8]
```diff
@@ -1114,11 +1127,14 @@ private function setRelationDependencies(
     /**
      * Creates a query instance for `has-one` or `has-many` relation.
      *
-     * @param string $arClass The class name of the related record.
+     * @param class-string<T> $arClass The class name of the related record.
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:1130]:
```suggestion
     * @param string $arClass The class name of the related record.
```

File: src/BaseActiveRecord.php [hunk 9]
```diff
@@ -1114,11 +1127,14 @@ private function setRelationDependencies(
...
      * @param array $link The primary-foreign key constraint.
      * @param bool $multiple Whether this query represents a relation to more than one record.
      *
      * @return ActiveQueryInterface The relational query object.
+     *
+     * @psalm-template T of ActiveRecordInterface
+     * @psalm-param class-string<T> $arClass
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:1137]:
```suggestion
     * @psalm-param class-string<ActiveRecordInterface> $arClass
```

**Review by Tigrov — CHANGES_REQUESTED** [review]

**Review by Tigrov — APPROVED** [review]
👍

**Pull request merged**